service/controller/resource/monitoring/remotewriteingressauth/resource.go

```diff
@@ -72,35 +76,61 @@ func toSecret(ctx context.Context, v interface{}, config Config) (*corev1.Secret
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
-	username := key.ClusterID(cluster)
 
-	config.Logger.Debugf(ctx, "generating password for the prometheus agent")
-	password, err := config.PasswordManager.GeneratePassword(32)
+	secretName, secretNamespace := key.RemoteWriteAPIEndpointConfigSecretNameAndNamespace(cluster, config.Provider)
+
+	config.Logger.Debugf(ctx, fmt.Sprintf("looking for secret %s in namespace %s", secretName, secretNamespace))
+	apiEndpointSecret, err := config.K8sClient.K8sClient().CoreV1().Secrets(secretNamespace).Get(ctx, secretName, metav1.GetOptions{})
 	if err != nil {
-		config.Logger.Errorf(ctx, err, "failed to generate the prometheus agent password")
+		config.Logger.Errorf(ctx, err, "failed to get api endpoint secret")
 		return nil, microerror.Mask(err)
 	}
 
-	hashedPassword, err := config.PasswordManager.Hash(password)
+	username, password, err := extractUsernameAndPasswordFromSecret(apiEndpointSecret)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	config.Logger.Debugf(ctx, "hashing password for the prometheus agent")
+	hashedPassword, err := config.PasswordManager.Hash([]byte(password))
 	if err != nil {
 		config.Logger.Errorf(ctx, err, "failed to hash the prometheus agent password")
 		return nil, microerror.Mask(err)
 	}
-	config.Logger.Debugf(ctx, "generate password for the prometheus agent")
+
+	config.Logger.Debugf(ctx, "hashed password for the prometheus agent")
 
 	secret := &corev1.Secret{
 		ObjectMeta: objectMeta,
 		Data: map[string][]byte{
-			// To be used by the ingress basic auth
-			"auth":     []byte(fmt.Sprintf("%s:%s", username, hashedPassword)),
-			"password": []byte(password),
-			"username": []byte(username),
+			"auth": []byte(fmt.Sprintf("%s:%s", username, hashedPassword)),
 		},
 		Type: "Opaque",
 	}
 	return secret, nil
 }
 
+func extractUsernameAndPasswordFromSecret(secret *corev1.Secret) (string, string, error) {
+	data := secret.Data["values"]
+	remoteWriteValues := remotewriteapiendpointconfigsecret.GlobalRemoteWriteValues{}
+	err := yaml.Unmarshal(data, &remoteWriteValues)
+	if err != nil {
+		return "", "", microerror.Mask(err)
+	}
+	if len(remoteWriteValues.Global.RemoteWrite) == 0 {
+		// skipping
+		return "", "", secretNotFound
+	}
+
+	username := remoteWriteValues.Global.RemoteWrite[0].Username
```

Comment:
Is it enough to get it from first item? 
I think we can check the name of RemoteWrite="prometheus"

Reply:
You are definitely right